packages/m2m-gateway/src/services/purposeService.ts

```diff
@@ -403,5 +408,40 @@ export function purposeServiceBuilder(clients: PagoPAInteropBeClients) {
       const polledPurpose = await pollPurposeById(purposeId, metadata, headers);
       return toM2MGatewayApiPurpose(polledPurpose.data);
     },
+    getPurposeAgreement: async (
+      purposeId: PurposeId,
+      { headers, logger }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.Agreement> => {
+      logger.info(`Retrieving agreement for purpose with id ${purposeId}`);
+
+      const { data: purpose } = await retrievePurposeById(purposeId, headers);
+
+      const { data: agreements } =
+        await clients.agreementProcessClient.getAgreements({
+          queries: {
+            consumersIds: [purpose.consumerId],
+            eservicesIds: [purpose.eserviceId],
+            states: [
+              m2mGatewayApi.AgreementState.Values.ACTIVE,
+              m2mGatewayApi.AgreementState.Values.SUSPENDED,
+            ],
+            offset: 0,
+            limit: 1,
+          },
```

Comment:
Consider extracting this into a function as I did for get agreement purposes here: https://github.com/pagopa/interop-be-monorepo/pull/2074/files#diff-60d8eba06154d053b1a1a97074a32e4f746ec522829b30eb0bb98acb481e3e6eR110

`toGetAgreementsApiQueryParamsForPurpose` or something similar